### PR TITLE
[HYDRATOR-1065] Removing unnecessary file tracking in XMLReader plugin

### PR DESCRIPTION
**Pull request by romy-khetan**

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java [hunk 1]
```diff
@@ -128,6 +98,31 @@ public LongWritable getCurrentKey() throws IOException, InterruptedException {
 
   @Override
   public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+    FileSplit fileSplit  = (FileSplit) split;
+    file = fileSplit.getPath();
+    fileName = file.toUri().toString();
+    Configuration conf = context.getConfiguration();
+    fs = file.getFileSystem(conf);
+    XMLInputFactory factory = XMLInputFactory.newInstance();
+    fdDataInputStream = fs.open(file);
```
Comment (CuriousVini) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java:108]: We should close the `fdDataInputStream`

**Review by CuriousVini — COMMENTED** [review]
Also we should test these code changes on cluster.

**Comment by romy-khetan** [comment]
@CuriousVini 
We have tested this fix on cluster. Please review.

**Comment by anwar6953** [comment]
LGTM

**Comment by abhinavbansal246** [comment]
Please squash commits and paste ITN results in the PR

**Comment by romy-khetan** [comment]
@abhinavbansal246 
Squashed the commits. ITN for XML reader plugin has been ignored due to FileSet issues, therefore couldn't paste ITN results in the PR.

**Pull request merged**